### PR TITLE
Automatically add extra options  for none driver on ubuntu

### PR DESCRIPTION
**Pull request by medyagh**
To fix this issue, https://github.com/kubernetes/minikube/issues/3511

### CODE REVIEW

File: cmd/minikube/cmd/start.go [hunk 1]
```diff
@@ -486,6 +497,16 @@ func generateConfig(cmd *cobra.Command, k8sVersion string) (cfg.Config, error) {
 	return cfg, nil
 }
 
+// noneNeedsResolveConf checks extra config option for resolv.conf is needed
+// for none driver. more info https://github.com/kubernetes/minikube/issues/3511
+func noneNeedsResolveConf() bool {
+	filename := "/run/systemd/resolve/resolv.conf"
```
Comment (tstromberg) [inline cmd/minikube/cmd/start.go]: Could this be refactored in a way that doesn't require this filename to be duplicated by the caller?

Perhaps this function could return the extra flag required (kubelet.resolv-conf=xxx), or the value could be shared as a const?

File: cmd/minikube/cmd/start.go [hunk 2]
```diff
@@ -350,6 +350,17 @@ func validateConfig() {
 		exit.Usage("Sorry, the --hidden feature is currently only supported with --vm-driver=kvm2")
 	}
 
+	// Adding extra option (resolv-conf) for none driver on ubutnu to address /issues/3511
+	if viper.GetString(vmDriver) == constants.DriverNone && noneNeedsResolveConf() {
+		o := "kubelet.resolv-conf=/run/systemd/resolve/resolv.conf"
+		err := extraOptions.Set(o)
+		if err == nil {
+			glog.Infof("Sucsessfully set extra option (%s) for none driver on ubuntu-like OS.", o)
```
Comment (tstromberg) [inline cmd/minikube/cmd/start.go]: Spelling: "Successfully". Also, no need to mention ubuntu-like OS. If you need to, it should be capitalized: "Ubuntu".

File: cmd/minikube/cmd/start.go [hunk 3]
```diff
@@ -350,6 +350,17 @@ func validateConfig() {
...
+		} else {
+			glog.Errorf("Error setting extra options (%s) for none driver on ubuntu-like OS: %s", o, err)
```
Comment (tstromberg) [inline cmd/minikube/cmd/start.go]: Shorten if possible. Just: "Error setting options %q: %v"

error types should always be %v instead of %s.

**Comment by medyagh** [comment]
@minikube-bot okay to test.

File: cmd/minikube/cmd/start.go [hunk 4]
```diff
@@ -486,6 +497,16 @@ func generateConfig(cmd *cobra.Command, k8sVersion string) (cfg.Config, error) {
 	return cfg, nil
 }
 
+// noneNeedsResolveConf checks extra config option for resolv.conf is needed
+// for none driver. more info https://github.com/kubernetes/minikube/issues/3511
+func noneNeedsResolveConf() bool {
+	filename := "/run/systemd/resolve/resolv.conf"
+	if _, err := os.Stat(filename); os.IsNotExist(err) {
```
Comment (tstromberg) [inline cmd/minikube/cmd/start.go]: optional: one could argue if there is any error calling os.Stat(), that we shouldn't use this file. I think you can safely remove the os.IsNotExist check.

Alternatively, you could also check for other more exceptional errors and log them.

**Comment by medyagh** [comment]
@minikube-bot okay to test

**Comment by medyagh** [comment]
DNS test failure is related to this issue : https://github.com/kubernetes/minikube/issues/4475

**Comment by RA489** [comment]
@medyagh Pleash squash the commits to one rest LGTM

**Pull request merged**